bftengine/src/bftengine/RequestsBatchingLogic.cpp

```diff
@@ -81,6 +84,29 @@ PrePrepareMsg *RequestsBatchingLogic::batchRequestsSelfAdjustedPolicy(SeqNum pri
   return replica_.buildPrePrepareMessage();
 }
 
+void RequestsBatchingLogic::adjustPreprepareSize() {
+  auto totalConsensusesCount = closedOnLogic + closedOnFlush;
+  auto perClosedOnFlush = (double)closedOnFlush / totalConsensusesCount;
+  auto perClosedOnLogic = (double)closedOnLogic / totalConsensusesCount;
+  start_timer_ = std::chrono::steady_clock::now();
+  if (perClosedOnFlush > 0.5 && maxNumOfRequestsInBatch_ > initialMaxNumOfRequestsInBatch_)
+    maxNumOfRequestsInBatch_ *= 0.8;
+  else if (perClosedOnFlush > 0.25 && maxNumOfRequestsInBatch_ > initialMaxNumOfRequestsInBatch_) {
+    maxNumOfRequestsInBatch_ *= 0.9;
+  } else if (maxNumOfRequestsInBatch_ <= 3 * initialMaxNumOfRequestsInBatch_) {
+    if (perClosedOnLogic >= 0.95) {
+      maxNumOfRequestsInBatch_ += maxNumOfRequestsInBatch_ * 0.15;
+    } else if (perClosedOnLogic >= 0.9) {
+      maxNumOfRequestsInBatch_ += maxNumOfRequestsInBatch_ * 0.1;
+    } else if (perClosedOnLogic >= 0.75) {
+      maxNumOfRequestsInBatch_ += maxNumOfRequestsInBatch_ * 0.05;
```

Comment:
Make all parameters configurable. It will speed up the tuning job.

Reply:
Which parameters? the increase rate or the increase condition?

Reply:
All the hard-coded numbers.

Reply:
Configuration parameters for increasing and decreasing rates, as well as conditions, have been added.